gatorgrade/output/output_functions.py

```diff
@@ -1,7 +1,57 @@
-"""Add docstring."""
+"""
+This class is used for storing the main functions requested from the Github.
 
+Issue Tracker for the output team.
+For instance, functions dealing with percentage output, description output,
+and colorization of text.
+"""
+import colorama as color
 
-# This class is used for storing the main functions requested
-# from the Github Issue Tracker for the output team.
-# For instance, functions dealing with percentage output,
-# description output, and colorization of text.
+color.init()
+
+
+def sort_checks_by_result(results):
+    """
+    Process results and determine if the check passed or failed.
+
+    Argument `results` is: list[(check_result_element1, check_result_element2),(...)]]

```

Comment:
There should be a couple extra parentheses here for the argument example that were forgotten:

Argument `results` is: list[(check_result_element1), (check_result_element2),(...)]]

Reply:
@gvanzin-allegheny @mccurdyc Unfortunately, this data structure (which is a tuple that looks like this: `(description: str, passed: bool, diagnostic: str)`) is provided by an outside tool (GatorGrader), so adding an object to represent it would add on additional translation overhead. This may be the perfect use case for a [namedtuple](https://docs.python.org/3/library/collections.html#collections.namedtuple), however!

Reply:
I wrote it the way I had it because I was trying to show how there are multiple elements in each check tuple with `(check_result_element1, check_result_element2)`, and that there are multiple checks in the result list by including the `,(...)` portion. However perhaps it could be rewritten to `list[(check_result_element1, check_result_element2, check_result_element3),(...)]`. What do you think?

Reply:
@lukel43 I think what you have in the comment is great! Could we use named tuples like @Michionlion suggested? I think that would help a ton from a readability perspective!

Not a huge deal if we can't use named tuples, just a strong suggestion! 😄

Reply:
Thanks, I will definitely update the comment, but I think it's best to not change the data structure at this time. A more descriptive comment, combined with the other clarity and readability improvements that were made should make the flow of data clear!